update-submodules.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+fail() {
+    printf '%s\n' "$1" >&2
+    exit "${2-1}"
+}
+
+# make sure we're in the root directory of the repository
+if [ ! -f "Cargo.toml" ]; then
+    fail "Failed to find \"Cargo.toml\", are you sure you're in the root directory of the repository?"
+fi
```

Comment:
`git ref-parse --show-toplevel` could be good here: that would return the top of the repository. From a quick search, some conditional like:

```bash
[[ $(git rev-parse --show-toplevel 2>/dev/null) = "$PWD" ]]
```

But if we're assured to have a `Cargo.toml` at the root of the repository (and no where else in the codebase), that should work too.